### PR TITLE
Added AMS; Added HOST env to rails config

### PR DESCRIPTION
**Pull request by vladimirbazhanov**
# Summary

- ActiveModelSerializer added, made serializer for User
- Added ENV["HOST"] for accepted hosts

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -7,4 +7,14 @@ class ApplicationController < ActionController::Base
 
   responders :flash
   respond_to :html
+
+  expose :serialized_current_user_data, -> { serialized_current_user_data }
```
Comment (DmitryBarskov) [inline app/controllers/application_controller.rb:11]:
```suggestion
  expose :serialized_current_user_data, :serialized_current_user_data
```

File: app/controllers/application_controller.rb [hunk 2]
```diff
@@ -7,4 +7,14 @@ class ApplicationController < ActionController::Base
...
+
+  private
+
+  def serialized_current_user_data
+    return nil unless user_signed_in?
+
+    UserSerializer.new(current_user).to_json
+  end
```
Comment (DmitryBarskov) [inline app/controllers/application_controller.rb:19]:
```suggestion
  def serialized_current_user_data
    UserSerializer.new(current_user).to_json if user_signed_in?
  end
```

File: app/controllers/application_controller.rb (hunk 1 above)
Comment (vladimirbazhanov) [reply app/controllers/application_controller.rb:11]: Does the second variant better then first?)

File: app/controllers/application_controller.rb (hunk 2 above)
Comment (vladimirbazhanov) [reply app/controllers/application_controller.rb:19]: I think "Explicit is better than implicit." I prefer to explicitly return values to easy see where and what I''ll get)

File: config/application.rb [hunk 3]
```diff
@@ -36,5 +36,8 @@ class Application < Rails::Application
 
     # Set URL options to be able to use url_for helpers
     config.action_mailer.default_url_options = { host: ENV.fetch("HOST") }
+
+    # Allowed hosts:
+    config.hosts << ["localhost", ENV.fetch("HOST")]
```
Comment (DmitryBarskov) [inline config/application.rb:41]: Why do we need `localhost` for all environments? It is included in development environment by default.

File: Gemfile [hunk 4]
```diff
@@ -25,6 +25,7 @@ gem "simple_form"
 gem "slim"
 
 # all other gems
+gem "active_model_serializers", "~> 0.10.0"
```
Comment (timurvafin) [inline Gemfile:28]: Do we need  `"~> 0.10.0"`?

File: app/controllers/application_controller.rb [hunk 5]
```diff
@@ -7,4 +7,12 @@ class ApplicationController < ActionController::Base
 
   responders :flash
   respond_to :html
+
+  expose :serialized_current_user_data, :serialized_current_user_data
```
Comment (timurvafin) [inline app/controllers/application_controller.rb:11]: Could we use `expose_decorated` to decorate `current_user` with serializer?

File: config/environments/test.rb [hunk 6]
```diff
@@ -45,4 +45,7 @@
   # config.action_view.raise_on_missing_translations = true
 
   config.cache_classes = false
+
+  # Allowed hosts:
+  config.hosts << "www.example.com"
```
Comment (timurvafin) [inline config/environments/test.rb:50]: Could we use `lvh.me`

**Pull request closed without merging**